### ISSUE TITLE
[Unsupported Feature] Cannot use `Default` trait of a tuple

### ISSUE THREAD

**Issue by kim-jhyeon** [kaist-cp/cs492-uarch#94]
The current HazardFlow compiler appears to be unable to handle the usage of `Default::default` for tuples in which all elements implement the `Default` trait.

For instance, the following code fails to compile:

```rust
// Note that both `U<3>` and `MeshTag` implement `Default`.
let a: (U<3>, MeshTag) = Default::default();
// ... use `a` afterward...
```

This produces an error similar to the following:

```text
thread 'rustc' panicked at hazardflow/src/compiler/pure/build_expr_ast.rs:762:39:
not yet implemented: (std::value::array::Array<bool, 3_usize>, gemmini::execute::systolic_array::mesh_with_delays::MeshTag)
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

note: Hazardflow has paniced!
  |
  = note: Please report this bug over here: https://github.com/kaist-cp/hazardflow
```

**Comment by jirheee** [comment]
First, thank you for reporting the bug!

We currently do not support using any functions outside the `hazardflow-designs` crate because the Rust compiler does not give direct access to ASTs of external crates.

Therefore, `Default::default` for tuple, which is implemented in the [`core`](https://doc.rust-lang.org/src/core/tuple.rs.html#116-123) crate, cannot be used.

Therefore, please manually call the `default` function twice in this case.

Sorry for the inconvenience, and thank you again for reporting the bug.

**Comment by jirheee** [comment]
@minseongg, Could you add this case at #53? It seems like I cannot edit it. Also, I think this should be merged with #33's case since it is also failing because automatic implementation of the `into` function from the `From` trait resides in the external crate.
